lib/jxl/modular/transform/enc_palette.cc

```diff
@@ -158,6 +160,17 @@ struct PaletteIterationData {
   }
 };
 
+struct pixel_hash {
```

Comment:
I don't like this hasher very much :) We should probably use a proper hasher.

Reply:
Agreed, it's probably a quite poor hash function. Best to keep it as simple as possible though — collisions are not as problematic here as in the e1 palette detection...

Reply:
Switched back to the (ordered) `set` and `map`, because it looks like the speedup wasn't coming from switching to unordered anyway (just using a map instead of linear search also does the job), and for some reason `unordered_map` was randomly causing `bad_alloc` in the CI and I can't be bothered to investigate what's going on there.

Speed and density results are similar to what I reported before, it's maybe even slightly faster this way.